subsys/net/ip/tcp.c

```diff
@@ -1884,6 +1884,24 @@ static int send_reset(struct net_context *context,
 	return ret;
 }
 
+static u16_t adjust_data_len(struct net_pkt *pkt, struct net_tcp_hdr *tcp_hdr,
+			     u16_t data_len)
+{
+	u8_t offset = tcp_hdr->offset >> 4;
```

Comment:
Why do we shift offset right by 4, just to shift it left 2 on every usage below (3 times)? At the very least, there should have been a comment telling in what units the `offset` is (words vs bytes, or whatever).